util/mSCGroup.py

```diff
@@ -0,0 +1,305 @@
+    	    varm=anndata.varm,
+    	    varp=anndata.varp,
+        )
+
+        if self.verbose:
+            print("  FINISH DECATEGORICALIZING")
+        return anndata
+
+    # ----------------------------------------------------------------
+    def write_tiledb_group(self, anndata):
+        if self.verbose:
+            print("  START  WRITING %s" % (self.uri))
+
+        # ----------------------------------------------------------------
+        # Must be done first, to create the parent directory
+        tiledb.group_create(self.uri)
+        base_group = tiledb.Group(self.uri, "w")
+
+        # ----------------------------------------------------------------
+        X_uri = self.write_X(anndata)
+        base_group.add(uri=X_uri, relative=False, name="X")
+
+        # ----------------------------------------------------------------
+        obs_uri = self.write_obs_or_var(anndata.obs, "obs")
+        base_group.add(uri=obs_uri, relative=False, name="obs")
+
+        var_uri = self.write_obs_or_var(anndata.var, "var")
+        base_group.add(uri=var_uri, relative=False, name="var")
+
+        # ----------------------------------------------------------------
+        if len(anndata.obsm.keys()) > 0:
+            obsm_uri = self.write_annotation_matrices(anndata.obsm, "obsm")
+            base_group.add(uri=obsm_uri, relative=False, name="obsm")
+
+        if len(anndata.varm.keys()) > 0:
+            varm_uri = self.write_annotation_matrices(anndata.varm, "varm")
+            base_group.add(uri=varm_uri, relative=False, name="varm")
+
+        if len(anndata.obsp.keys()) > 0:
+            obsp_uri = self.write_annotation_matrices(anndata.obsp, "obsp")
+            base_group.add(uri=obsp_uri, relative=False, name="obsp")
+
+        if len(anndata.varp.keys()) > 0:
+            varp_uri = self.write_annotation_matrices(anndata.varp, "varp")
+            base_group.add(uri=varp_uri, relative=False, name="varp")
+
+        # ----------------------------------------------------------------
+        if self.verbose:
+            print("  FINISH WRITING %s" % (self.uri))
+
+        base_group.close()
+
+    # ----------------------------------------------------------------
+    def write_X(self, anndata):
+        X_uri = os.path.join(self.uri, "X")
+        tiledb.group_create(X_uri)
+        X_group = tiledb.Group(X_uri, "w")
+
+        X_data_uri = os.path.join(X_uri, "data")
+        if self.verbose:
+            print("    START  WRITING %s" % (X_data_uri))
+
+        # Here we do not use tiledb.from_numpy, so that we can have more control over the schema.
+
+        obs_dim, var_dim = np.meshgrid(anndata.obs.index, anndata.var.index)
+
+        dom = tiledb.Domain(
+            tiledb.Dim(name="obs_id", domain=(None, None), dtype="ascii"),
+            tiledb.Dim(name="var_id", domain=(None, None), dtype="ascii"),
+        )
+        att = tiledb.Attr("data")
+        sch = tiledb.ArraySchema(domain=dom, attrs=(att,), sparse=True)
```

Comment:
I think you have to add `allows_duplicates=True` as an argument here [to match the schema that was shown above](https://tiledb.slack.com/archives/C03BY1K3GP9/p1650821218628999?thread_ts=1650813691.071909&cid=C03BY1K3GP9).